yb-voyager/cmd/config.go

```diff
@@ -0,0 +1,467 @@
+	    - Additionally, handles special prefixed keys for source/target DB configs:
+	    - Flags starting with "source-" → looks under "source.<flag-suffix>"
+	    - Flags starting with "oracle-" → looks under "source.<oracle-flag-name>"
+	    - Flags starting with "target-" → looks under "target.<flag-suffix>"
+	 4. If a value is found in Viper, the corresponding flag is set with that value.
+	 5. If any error occurs during binding, it stops further processing and returns the error.
+
+	This function allows users to configure flags through the config file or environment variables,
+	while still letting command-line input take precedence.
+*/
+func bindCobraFlagsToViper(cmd *cobra.Command, v *viper.Viper) error {
+	var bindErr error
+
+	subCmdPath := strings.TrimPrefix(cmd.CommandPath(), cmd.Root().Name())
+	subCmdPath = strings.TrimSpace(subCmdPath) // remove leading space if any
+	// Replace spaces with hyphens
+	configKeyPrefix := strings.ReplaceAll(subCmdPath, " ", "-")
+	configKeyPrefix = setToAliasPrefixIfSet(configKeyPrefix, v)
+
+	cmd.Flags().VisitAll(func(f *pflag.Flag) {
+		if bindErr != nil {
+			return // Stop further execution if an error is already encountered
+		}
+
+		if !f.Changed {
+			// Check for <command_path>.<flagname>
+			if v.IsSet(configKeyPrefix + "." + f.Name) {
+				val := v.GetString(configKeyPrefix + "." + f.Name)
+				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
+				if err != nil {
+					// In case of an error while setting the flag from viper, return the error
+					bindErr = err
+					return
+				}
+			} else if v.IsSet(f.Name) {
+				// Bind the global flag from viper to cmd
+				val := v.GetString(f.Name)
+				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
+				if err != nil {
+					bindErr = err
+					return
+				}
+			} else if strings.HasPrefix(f.Name, "source-") && v.IsSet("source."+strings.TrimPrefix(f.Name, "source-")) {
+				// Handle source db type flags
+				val := v.GetString("source." + strings.TrimPrefix(f.Name, "source-"))
+				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
+				if err != nil {
+					bindErr = err
+					return
+				}
+			} else if strings.HasPrefix(f.Name, "oracle-") && v.IsSet("source."+f.Name) {
+				// Handle oracle db type flags, since they are also prefixed with source but are special cases
+				val := v.GetString("source." + f.Name)
+				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
+				if err != nil {
+					bindErr = err
+					return
+				}
+			} else if strings.HasPrefix(f.Name, "target-") && v.IsSet("target."+strings.TrimPrefix(f.Name, "target-")) {
+				// Handle target db type flags
+				val := v.GetString("target." + strings.TrimPrefix(f.Name, "target-"))
+				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
+				if err != nil {
+					bindErr = err
+					return
+				}
+				// fmt.Printf("binding %s from viper to cmd flag: %s=%s\n", "target."+strings.TrimPrefix(f.Name, "target-"), f.Name, val)
```

Comment:
add sufficient logs for every case. 
if we are ignoring a flag, because it was set in CLI
if we found it in global, if we found in cmd.key/ source.key/target-key.

Reply:
Ah i see. 
Let's discuss this in standup to see if we can come up with something.

Reply:
As discussed, pls collect all the config keys, and log them after logging is initialized.

Reply:
Done